rascal-lsp/src/main/java/org/rascalmpl/vscode/lsp/parametric/ParametricTextDocumentService.java

```diff
@@ -292,7 +309,7 @@ public CompletableFuture<List<? extends CodeLens>> codeLens(CodeLensParams param
         final ILanguageContributions contrib = contributions(params.getTextDocument());
 
         return recoverExceptions(file.getCurrentTreeAsync()
-            .thenApply(contrib::lenses)
+            .thenApply(t -> contrib.lenses(t.get()))
```

Comment:
NP: if we accept a 2nd `thenApply` we can skip the extra closure getting allocated. (same for the other places that have this)

Reply:
Fixed